service/src/main/kotlin/fi/espoo/evaka/messaging/MessageController.kt

```diff
@@ -282,66 +284,74 @@ class MessageController(
-                    municipalAccountName = featureConfig.municipalMessageAccountName
-                )
-                if (body.draftId != null)
-                    tx.deleteDraft(accountId = accountId, draftId = body.draftId)
-                messageContentId
+                    if (messageRecipients.isEmpty()) return@transaction null
+
+                    val staffCopyRecipients =
+                        if (body.recipients.none { it.type == MessageRecipientType.CHILD }) {
+                            tx.getStaffCopyRecipients(
+                                accountId,
+                                body.recipients.mapNotNull {
+                                    if (it.type == MessageRecipientType.AREA) AreaId(it.id.raw)
+                                    else null
+                                },
+                                body.recipients.mapNotNull {
+                                    if (it.type == MessageRecipientType.UNIT) DaycareId(it.id.raw)
+                                    else null
+                                },
+                                body.recipients.mapNotNull {
+                                    if (it.type == MessageRecipientType.GROUP) GroupId(it.id.raw)
+                                    else null
+                                },
+                                clock.today()
+                            )
+                        } else {
+                            setOf()
+                        }
+
+                    val groupedRecipients =
+                        tx.groupRecipientAccountsByGuardianship(messageRecipients)
+                    val messageContentId =
+                        messageService.createMessageThreadsForRecipientGroups(
+                            tx,
+                            clock,
+                            title = body.title,
+                            content = body.content,
+                            sender = accountId,
+                            type = body.type,
+                            urgent = body.urgent,
+                            recipientNames = body.recipientNames,
+                            recipientGroups = groupedRecipients,
+                            attachmentIds = body.attachmentIds,
+                            staffCopyRecipients = staffCopyRecipients,
+                            accountIdsToChildIds = messageAccountIdsToChildIds,
+                            municipalAccountName = featureConfig.municipalMessageAccountName
+                        )
+                    if (body.draftId != null)
+                        tx.deleteDraft(accountId = accountId, draftId = body.draftId)
+                    messageContentId
+                }
             }
-        }.also { Audit.MessagingNewMessageWrite.log(targetId = accountId, objectId = it) }
+            .also { Audit.MessagingNewMessageWrite.log(targetId = accountId, objectId = it) }
```

Comment:
Is the whole hunk just a formatting change, or are there some real changes I can't spot? I thought reformatting wouldn't be a problem now that we have `ktfmt` 😂

Reply:
The result of `db.conntect` is returned and the audit call if wrapped in a `also` block because of that. That's why the indentation is changed. I guess github doesn't notice that. I think ktfmt's formatting is a bit ugly here, but it should be consistent.